daemon/daemon.go

```diff
@@ -527,6 +529,7 @@ func (d *Daemon) Stop(sigCh chan<- os.Signal) error {
 	restartSystem := d.restartSystem
 	restartSocket := d.restartSocket
 	d.mu.Unlock()
+	immediate := restartSystem == state.RestartSystemImmediate
```

Comment:
this should move closer to where it's used I thin.  Or we move it inside  the locked section like:

```
restartSystem := d.restartSystem != state.RestartUnset
immediate :=  d.restartSystem == state.RestartSystemImmediate
```